doc/exts/pylint_messages.py

```diff
@@ -158,32 +158,58 @@ def _get_all_messages(
             checker_module.__name__,
             checker_module.__file__,
         )
-        messages_dict[MSG_TYPES_DOC[message.msgid[0]]].append(message_data)
-
+        msg_type = MSG_TYPES_DOC[message.msgid[0]]
+        messages_dict[msg_type].append(message_data)
         if message.old_names:
             for old_name in message.old_names:
                 category = MSG_TYPES_DOC[old_name[0][0]]
                 old_messages[category][(old_name[1], old_name[0])].append(
-                    (message.symbol, MSG_TYPES_DOC[message.msgid[0]])
+                    (message.symbol, msg_type)
                 )
 
     return messages_dict, old_messages
 
 
+def message_needs_update(message_data: MessageData, category: str) -> bool:
+    """Do we need to regenerate this message .rst ?"""
+    message_path = get_message_path(category, message_data)
+    if not message_path.exists():
+        return True
+    message_path_stats = message_path.stat().st_mtime
+    checker_path_stats = Path(message_data.checker_module_path).stat().st_mtime
+    return checker_path_stats > message_path_stats
+
+
+def get_category_directory(category: str) -> Path:
+    return PYLINT_MESSAGES_PATH / category
+
+
+def get_message_path(category: str, message: MessageData) -> Path:
+    category_dir = get_category_directory(category)
+    return category_dir / f"{message.name}.rst"
+
+
 def _write_message_page(messages_dict: MessagesDict) -> None:
     """Create or overwrite the file for each message."""
     for category, messages in messages_dict.items():
-        category_dir = PYLINT_MESSAGES_PATH / category
+        category_dir = get_category_directory(category)
         if not category_dir.exists():
             category_dir.mkdir(parents=True, exist_ok=True)
         for message in messages:
-            checker_module_rel_path = os.path.relpath(
-                message.checker_module_path, PYLINT_BASE_PATH
-            )
-            messages_file = os.path.join(category_dir, f"{message.name}.rst")
-            with open(messages_file, "w", encoding="utf-8") as stream:
-                stream.write(
-                    f""".. _{message.name}:
+            if not message_needs_update(message, category):
+                print(f"{message.name} is up to date.", end="\r")
```

Comment:
Shall we log this? I haven't tested locally, but doesn't this feel spammy?

Reply:
The ``, end="\r")`` always rewrite the same line. ``\r`` is a carriage return so it behave like a typing machine where you don't provide a line feed (it's the reason why windows end of line are/can be``\r\n`` and not ``\n``, really legacy stuff.)

Reply:
Smart, learned a lot from this PR 🎉

Reply:
Fun fact originally each OS had chosen a different line ending convention.
- Windows : carriage return then line feed (most historically correct regarding what you really do on a typing machine)
- Mac: carriage return (would rewrite the current line)
- Linux : line feed (would write on the next line but starting from the column of the last char)